### PR TITLE
fix: Fix animation setters not executed if a transition contains RepositionThemeAnimation

### LINKED ISSUE #13303: [ToggleSwitch/GTK/WASM] Thumb in wrong position when adding Shadows
### Current behavior

While working with ShadowContainer and adding ToggleSwitch styles I came across this problem where the Thumb gets in the wrong position on GTK and WASM. It works on Windows. 

I isolated my style in the following repro:

[ToggleSwitchShadowRepro.zip](https://github.com/unoplatform/uno/files/12381777/ToggleSwitchShadowRepro.zip)

Please don't mind the project name and the mixed colors.

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #13303

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/Animation/RepositionThemeAnimation.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[global::Uno.NotImplemented]
+#endif
+public partial class RepositionThemeAnimation : Timeline, ITimeline
+{
+	// RepositionThemeAnimation is currently not implemented.
+	// But we need to call OnCompleted() so that whenever RepositionThemeAnimation is
+	// included in a Storyboard, we want to allow the Storyboard to fire Completed event
+	// This is important especially when RepositionThemeAnimation is part of a transition Storyboard.
+	// Visual state setters aren't executed unless the transition is completed.
+	void ITimeline.Begin() => OnCompleted();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Media/Animation/RepositionThemeAnimation.cs:13]: That's an interesting workaround. There are other animations like this one that likely would benefit from the same change (in another PR).

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Media/Animation/RepositionThemeAnimation.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Media/Animation/RepositionThemeAnimation.cs:13]: Yeah I thought of other animations, but found `RepositionThemeAnimation` to be the most commonly used one and wanted to quickly get #13303 fixed. Other non-implemented animations are very rarely used. We can follow-up if needed in another PR.

**Comment by agneszitte** [comment]
Re-Open for backport PR

**Comment by Youssef1313** [comment]
@mergifyio backport legacy/4x

**Pull request merged**